### PR TITLE
MUMUP-1570 - Widget MVP: Toggle and Min Mode

### PR DESCRIPTION
**Pull request by jhanstra**
There were several ideas for how to approach the development of the widget-view/content-on-home/minimized-portlets project that we discussed in grooming a couple weeks ago. At the time of that discussion, I had a git branch that I was using to get started on the implementation and test some things out. I added more to that branch and it's looking pretty good, I think. This is that branch.

In this pull request is a working toggle to switch between a list view and a not-yet-implemented widget view. The list view implements @keirserrie's mockups for a minimized view. I am pleased with how it turned out and I propose that we use this view as the new minimized view - in other words, we don't go down the road where we have 3 views. This is just my personal opinion, but I think this is a pretty good MVP for minimized view and it didn't take long to build at all. Instead of rolling out new views twice, I think we could replace the current view with this min view and add a second view for widgets. Thoughts? 

Notes for review:
- myuw-toolkit-temporary.less contains styles that are in an unreleased version of the toolkit. As soon as we release a new version with all the new goodies we'll delete the styles in this file.
- I used the inner-nav from the toolkit on the home page to be consistent. I also put the toggle in said inner-nav instead of the top right of the cover photo because I think it makes more sense there.
- The toggle is behind a beta flag, but the new styling for minimized view is not.

### Screenshots
image

image

Placeholder for widget view:
image

Hover state:
image

Mobile (for now):
image

For reference, here's @keirserrie's mockup:
image

A few things still to do (but not sure if they should all go in this pull request because it's already big):
- Optimize for mobile -- make it more listy less boxy
- Add remove button on hover for ability to remove an app
- "Add Apps" button as the last item, taking you to marketplace
- Give area a max width so enormous screens don't have stretched out boxes

### CODE REVIEW

**Comment by apetro** [comment]
One of the most important things to say about this changeset is that it is beautiful work.  So let me go ahead and say that here, and let's have a nuanced conversation about Scrum face to face. :smile:

**Comment by keirserrie** [comment]
Looking really good..wasn't expecting this to come on Thursday! :)

File: angularjs-portal-home/src/main/webapp/partials/default-card.html [hunk 1]
```diff
@@ -1,16 +1,13 @@
-<div class="portlet-div" id="portlet-id-{{::portlet.nodeId}}">
-
-  <a href="{{::portlet.url}}" target="{{::portlet.target}}" >
-    <div class="portlet-title">
-        <portlet-icon></portlet-icon><h1>{{ ::portlet.title }}</h1>
+<div class="list-content" id="portlet-id-{{::portlet.nodeId}}">
+  <a href="{{::portlet.url}}" target="{{::portlet.target}}">
+    <div class="icon-container">
+      <i class="fa {{::portlet.faIcon}}" ng-if="portlet.faIcon !== '' && portlet.faIcon !== null"></i>
```
Comment (timlevett) [inline angularjs-portal-home/src/main/webapp/partials/default-card.html]: Could we use the `portlet-icon` directive here instead?

File: angularjs-portal-home/src/main/webapp/partials/layout.html [hunk 2]
```diff
@@ -20,20 +20,22 @@ <h3 class="home-section-header">Notifications</h3>
...
-           popover-placement="top"
-           popover-popup-delay="500">
-            <i class='fa fa-plus'></i>
-        </a>
+    <div class="inner-nav-container">
+      <ul class="inner-nav">
+        <li class="active">
+            <a>Your Favorites</a>
+        </li>
+        <span class="layout-chooser" ng-if="$storage.contentView">
```
Comment (timlevett) [inline angularjs-portal-home/src/main/webapp/partials/layout.html]: this if should be checking `$scope.toggle` that was created in MUMUP-1609

File: angularjs-portal-home/src/main/webapp/partials/default-card.html [hunk 3]
```diff
@@ -1,16 +1,13 @@
...
-  <a href="{{::portlet.url}}" target="{{::portlet.target}}" >
-    <div class="portlet-content">
-      {{:: portlet.description | truncate:160}}
+    <div class="list-item-container">
+      <h4>{{ ::portlet.title }}</h4>
     </div>
   </a>
+  <!-- <div class='card-remove'>
+    <i title="Remove" class="fa fa-times portlet-options" ng-click="layoutCtrl.removePortlet(portlet.nodeId, portlet.title)"></i>
+  </div> -->
```
Comment (timlevett) [inline angularjs-portal-home/src/main/webapp/partials/default-card.html]: We need to be able to remove before this is merged.

File: angularjs-portal-home/src/main/webapp/partials/settings.html [hunk 4]
```diff
@@ -79,6 +79,16 @@ <h3 class="home-section-header">Your Options</h3>
             </h4>
             <small>Enable/Disable the typeahead in the portlet search bar</small>
         </li>
+        <li class="portlet-container-home beta-card-style">
```
Comment (timlevett) [inline angularjs-portal-home/src/main/webapp/partials/settings.html]: Not needed.

**Comment by jhanstra** [comment]
Updated to address all @timlevett's comments, along with:
- The ability to remove apps (shows on hover)
- Ability to view description (info icon shows on hover)
- Mobile looks better - app tiles are smaller and in 2 columns
- On super huge screens there is a max-width on the grid so that the portlets don't stretch to a crazy extent
- Add Apps placeholder taking you to marketplace

It should be fully functional and ready to ship, so let me know if there is anything I missed.

More screenshots:
image

Hover state:
image

Viewing description:
image

Mobile:
image

Super huge:
image

**Comment by vertein** [comment]
I like that it scales up well.

**Comment by vertein** [comment]
:+1:

**Comment by timlevett** [comment]
:+1:

**Comment by timlevett** [comment]
Looks awesome @jhanstra . Well done!

**Pull request merged**